lib/matplotlib/pyplot.py

```diff
@@ -995,18 +995,24 @@ def figure(
     fig_label = ''
     if num is None:
         num = next_num
-    elif isinstance(num, str):
-        fig_label = num
-        all_labels = get_figlabels()
-        if fig_label not in all_labels:
-            if fig_label == 'all':
-                _api.warn_external("close('all') closes all existing figures.")
-            num = next_num
-        else:
-            inum = all_labels.index(fig_label)
-            num = allnums[inum]
     else:
-        num = int(num)  # crude validation of num argument
+        if any([figsize, dpi, facecolor, edgecolor, not frameon,
+                kwargs]) and num in allnums:
```

Comment:
Reading this one more time, I think we also need the warning around line 988 for the case were someone does:

```python
fig = plt.figure()
plt.figure(fig, figsize=(2, 2))  # this should warn as well
```

Reply:
Yep, added that as well